src/Microsoft.DotNet.Build.Tasks/Targets/packages.targets

```diff
@@ -48,7 +48,7 @@
       Text="%(PackagesNuSpecFiles.Filename) NuGet Package -> $(PackagesOutDir)%(PackagesNuSpecFiles.Filename).%(PackagesNuSpecFiles.PackageVersion).nupkg" />
 
     <ItemGroup>
-      <PackagesForPublishing Include="$(PackagesOutDir)\*.nupkg" />
+      <PackagesForPublishing Include="$(PackagesOutDir)*.nupkg" />
     </ItemGroup>
```

Comment:
One common approach to avoid these things is to just check for the trailing slash and if it doesn't exist add it.